packages/ui/src/InteractiveList.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { FocusEvent, Ref, useCallback, useImperativeHandle, useMemo, useState } from 'react'
+import { TextField, TextFieldExtraProps } from './TextField'
+import { IconButton } from './IconButton'
+import { Plus, X } from 'react-feather'
+import styles from './InteractiveList.module.scss'
+import { useUID } from 'react-uid'
+import { FieldArrayRenderProps, FieldValidator } from 'formik'
+import { Error, errorId } from './Error'
+
+export type InteractiveListExtraProps = {
+  children?: React.ReactNode | React.ReactNode[]
+  inputControlRef?: Ref<InteractiveListInputRef>
+} & Pick<TextFieldExtraProps<'text'>, 'label' | 'helperText' | 'inputIcon' | 'type' | 'placeholder'>
+
+export type InteractiveListCoreProps = {
+  name: string
+  value: string[]
+  onBlur?: (e: FocusEvent<HTMLTextAreaElement>) => void
+  error?: string
+  validate?: FieldValidator
+  arrayHelpers: FieldArrayRenderProps
+  onError: (str: string) => void
+} & InteractiveListExtraProps
+
+export type InteractiveListProps = InteractiveListExtraProps & InteractiveListCoreProps
+
+export type InteractiveListInputRef = { setValue: (value: string) => void }
+
+export type InteractiveListItemProps = {
+  arrayHelpers: FieldArrayRenderProps
+  content: string
+  error?: string
+  idx: number
+}
+
+export const InteractiveListItem = ({ arrayHelpers, content, error, idx }: InteractiveListItemProps) => {
+  const id = useUID()
+  const errorProps = error
+    ? {
+        'aria-invalid': true,
+        'aria-errormessage': errorId(id),
+      }
+    : {}
+
+  return (
+    <>
+      <div className={styles.content}>
+        <span id={id} {...errorProps}>
+          {content}
+        </span>{' '}
+        <IconButton
+          kind="transparent"
+          ariaLabel="Remove Item"
+          icon={X}
+          size="small"
+          onClick={() => {
+            arrayHelpers.remove(idx)
+          }}
+        />
+      </div>
+      {error && <Error error={error} inputId={id} />}
+    </>
+  )
+}
+
+/**
+ * This component is meant to be used only from InteractiveListFormik
+ * @private
+ */
+const InteractiveList = ({
+  label,
+  inputControlRef,
+  children,
+  error,
+  name,
+  value,
+  helperText,
+  inputIcon,
+  onError,
+  type,
+  validate,
+  arrayHelpers,
+}: InteractiveListProps) => {
+  const id = useUID()
+  const [inputValue, setValue] = useState<string>('')
+
+  const normalizedValue = useMemo(() => inputValue.trim(), [inputValue])
+
+  useImperativeHandle(inputControlRef, () => ({
+    setValue: (value: string) => {
+      setValue(value)
+    },
+  }))
+
+  const addValueCallback = useCallback(async () => {
+    if (normalizedValue.length === 0) {
+      onError('You need to provide a non empty value')
+    } else if (value.includes(normalizedValue)) {
+      onError('You need to provide a unique value')
+    } else {
```

Comment:
@jvorcak should this be a part of `validate`?

Reply:
Since we can display the erro in a single place only - under the top input, should we also change our stories to always return a string or undefined from the validator? In other words, what sense does it make to returned structured errors for every item in the array when we do not diaply them? Moreover, shouldn't;t we rather validate our input value before we add it to the list, so there are no errors in the list ever?

Reply:
@aigoncharov this is the problematic part. We show errors also per item, this is simply a requirement from the Formik side in case we use FieldArray with it's helpers. The reason for this is that Formik has built-in `yup` validation mechanism, and that can provide your FieldArray component with an array of errors, the component should support it. There is no other way around it except for forbidding Yup completely somehow. 

The problematic part is that there is no way (or I haven't found any) how to run a `Yup` validation for this specific field on demand by providing a value (from our custom field) and getting a result (that could prevent from entering a value).

So this solution is a fallback, if you provide validate, we will guard you on entering the value, and support `yup` validation as a last check before submitting the value -> so basically the UX is not ideal if Yup validation contains more checks than custom validate. This is what I tried to point out initially, hopefully it's more clear now.

Reply:
@jvorcak do you use schema-based validation in Cloud?

Reply:
And with Yup, you'll need to make sure to provide it explicitelly via validate to get a better UX. Not exactly ideal, but the best I could think of with Formik

Reply:
@jvorcak maybe, I am overthinking this. Let's stick to the current implementation. One thing though. It does not re-validate the input value on change for me in storybook. Could you check that?

Reply:
I don't think you are, I'm not 100% happy with it either, but I've spent enough time on investigating how it could be done better, and this is the best I could think of.